Automattic/jetpack
# VideoPress: Add videopress embed shortcode

Fixes #27841

#### Changes proposed in this Pull Request:

* Adds `videopress` and `wpvideo` shortcodes, overriding Jetpack's shortcodes
* Adds a shortcode quick copy field to the video details page

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Go to the VideoPress dashboard without any video
* Upload one video and check that the "Add video to post" feature still works as expected by clicking on `Add video to post` and verifying that the block is rendered:

![2022-12-21_10-36-39](https://user-images.githubusercontent.com/8486249/208921406-5afbb7a5-fb89-4cad-b002-73dbda34bdca.png)

* Change the newly uploaded video's privacy to `private`:

![2022-12-21_10-37-16](https://user-images.githubusercontent.com/8486249/208921693-2613fbfc-b822-4c00-b076-7cbd48069c15.png)

* Click on `Edit video details` and copy the generated shortcode:

![2022-12-21_10-37-38](https://user-images.githubusercontent.com/8486249/208921842-70505b92-a3b3-4741-91e3-9c593b8bafac.png)

* Back on the block editor, add the generated shortcode to a `Shortcode` block:

![2022-12-21_10-38-00](https://user-images.githubusercontent.com/8486249/208922039-a1202534-7628-47be-bb8d-ada2c69fc399.png)

* Also add other variations, including the `video` shortcode, that is overriden:

![2022-12-21_10-48-30](https://user-images.githubusercontent.com/8486249/208922408-a34f9db5-9136-41e4-b305-cc2e5d25b153.png)

1 - The generated shortcode
2 - `videopress` (or `wpvideo`) shortcode with variations
3 - `video` shortcode with `videopress_guid` key
4 - `video` shortcode with `src`

* Publish and visit the post
* Check that all shortcodes are rendered correctly
* Visit the post in a new private window (incognito mode or in another browser where you are not logged in)
* Check that the videos are marked as private as expected:

![2022-12-08_16-53-45](https://user-images.githubusercontent.com/8486249/206554427-b2cd383b-0a88-4560-aed3-f94d883b3cd1.png)

* Install the Jetpack plugin
* In another tab, visit the post (VideoPress + Jetpack)
* Uninstall VideoPress
* In another tab, visit the post (Jetpack alone)
* Reinstall VideoPress
* In another tab, visit the post (Jetpack + VideoPress, installation order matters)
* Check that all four tabs were rendered without difference

![Screenshot 2022-12-21 at 11-08-20 Testing – HelloWord](https://user-images.githubusercontent.com/8486249/208925119-60a38972-1a5a-4446-a392-f262a890a5f9.png)
